### PR TITLE
node: make ipa-client available as optional

### PR DESCRIPTION
**Pull request by sandrobonazzola**
Allowing to install ipa-client on oVirt Node as it's not installed by default anymore

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=2017681
Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>

### CODE REVIEW

File: node-optional.el8.repo [hunk 1]
```diff
@@ -6,12 +6,58 @@ mirrorlist = http://mirrorlist.centos.org/?release=$stream&arch=$basearch&repo=B
 gpgcheck = 1
 enabled = 1
 gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
-includepkgs = dnf-plugin-subscription-manager fcoe-utils python3-librepo subscription-manager usermode
+includepkgs =
```
Comment (mykaul) [inline node-optional.el8.repo:9]: nit: can we sort this alphabetically?

File: configure.ac [hunk 2]
```diff
@@ -30,7 +30,7 @@ AC_SUBST([PACKAGE_RPM_VERSION])
 PACKAGE_SHORT_VERSION="$(echo VERSION_NUMBER | cut -d '.' -f 1,2,3)"
 AC_SUBST([PACKAGE_SHORT_VERSION])
 
-PACKAGE_RPM_RELEASE="0.0.$(echo VERSION_SUFFIX | sed 's/^_//')"
+PACKAGE_RPM_RELEASE="0.1.$(echo VERSION_SUFFIX | sed 's/^_//')"
```
Comment (didib) [inline configure.ac:33]: Is this really needed?

File: node-optional.el8.repo (hunk 1 above)
Comment (didib) [reply node-optional.el8.repo:9]: I'd personally actually do the opposite: Order in order of addition, and add comments (if possible) - e.g. "# ipa dependencies".

File: node-optional.el8.repo [hunk 3]
```diff
@@ -6,12 +6,58 @@ mirrorlist = http://mirrorlist.centos.org/?release=$stream&arch=$basearch&repo=B
...
+ sssd-tools
 
 [onn-appstream]
 name = oVirt Node Optional packages from CentOS Stream $releasever - AppStream
 mirrorlist = http://mirrorlist.centos.org/?release=$stream&arch=$basearch&repo=AppStream&infra=$infra
 gpgcheck = 1
 enabled = 1
 gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
-includepkgs = vhostmd
+includepkgs =
```
Comment (didib) [inline node-optional.el8.repo:45]: (same here)

File: node-optional.el9.repo [hunk 4]
```diff
@@ -2,16 +2,31 @@
 
 [onn-baseos]
 name = oVirt Node Optional packages from CentOS Stream $releasever - BaseOS
-baseurl = http://mirror.stream.centos.org/9-stream/BaseOS/$basearch/os/
+metalink = https://mirrors.centos.org/metalink?repo=centos-baseos-$stream&arch=$basearch&protocol=https,http
```
Comment (didib) [inline node-optional.el9.repo:5]: Perhaps add something about this to the commit message, if you do not want to split to its own patch

File: node-optional.el9.repo [hunk 5]
```diff
@@ -2,16 +2,31 @@
...
+gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
 gpgcheck = 1
+countme=1
```
Comment (didib) [inline node-optional.el9.repo:8]: whitespace around '=', keeping style? Also elsewhere.

File: node-optional.el9.repo [hunk 6]
```diff
@@ -2,16 +2,31 @@
...
+ ipa-client
+ ipa-client-common
+ ipa-common
+ ipa-selinux
 
 [onn-sap]
 name = oVirt Node Optional packages from CentOS Stream $releasever - SAP
 baseurl = https://composes.stream.centos.org/production/latest-CentOS-Stream/compose/SAP/$basearch/os/
-gpgcheck = 0
+gpgkey = file:///etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
```
Comment (didib) [inline node-optional.el9.repo:29]: Also add this to commit message

File: configure.ac (hunk 2 above)
Comment (sandrobonazzola) [reply configure.ac:33]: it help ensuring the update is visible but given we are building with a time stamp we can live without it.

File: node-optional.el8.repo (hunk 1 above)
Comment (sandrobonazzola) [reply node-optional.el8.repo:9]: I'll try making it grouped by reason and sorted alphabetically within the group.

File: node-optional.el8.repo (hunk 3 above)
Comment (sandrobonazzola) [reply node-optional.el8.repo:45]: same here

File: node-optional.el9.repo (hunk 4 above)
Comment (sandrobonazzola) [reply node-optional.el9.repo:5]: I'll move to separate patch

File: node-optional.el9.repo (hunk 5 above)
Comment (sandrobonazzola) [reply node-optional.el9.repo:8]: ok

File: node-optional.el9.repo (hunk 6 above)
Comment (sandrobonazzola) [reply node-optional.el9.repo:29]: moving to separate patch

**Comment by sandrobonazzola** [comment]
This requires pr #39 to be merged first

**Comment by didib** [comment]
> This requires pr #39 to be merged first

Strictly speaking, it does not - already discussed in private a few days ago, re the difference from gerrit - merging current will also merge the content of #39, which is one of the patches here. What you probably meant to say is something like:

#39 can/should be reviewed and merged independently of current. If so, current will then become just a single patch. Sadly, github does not reveal this nicely.

**Review by didib — APPROVED** [review]
